src/main/java/net/sf/jabref/model/entry/BibEntry.java

```diff
@@ -379,6 +372,42 @@ public void clearField(String name) {
     }
 
     /**
+     * Hide single chosen optional Field
+     * @param name field name has to be checked
+     */
+    public void toggleOptionalFieldVisibility(String name) {
```

Comment:
Unfortunately, this is not what is requested at https://github.com/JabRef/jabref/issues/759

I want to select a single field in the UI to hide or unhide (second bullet point).

The third bullet point means that I want to unhide all fields in the cleanup dialog. This is an request, which should come at last, the coloring and the hide/unhide functionality is more important.